stl/inc/regex

```diff
@@ -3910,6 +3910,10 @@ int _Parser<_FwdIt, _Elem, _RxTraits>::_Do_digits(int _Base, int _Count) { // tr
     _Val = 0;
     while (_Count != 0 && (_Chv = _Traits.value(_Char, _Base)) != -1) { // append next digit
         --_Count;
+        if (_Val > ((INT_MAX - _Chv) / _Base)) {
+            _Val = INT_MAX;

```

Comment:
Why not directly call `_Error(regex_constants::error_backref);`

Reply:
I think this part is used in multiple contexts; see unskipped libc++ tests catching a different exception

Reply:
I did it there: https://github.com/microsoft/STL/pull/2169/commits/30b4de669a19c716acdf2a4239ba39c2afb8629d
But tests fails.

Sometimes we need different exceptions.

https://github.com/microsoft/STL/blob/bc2c6b80dcaf6b10466cfb6f61fa32bcf70680ad/stl/inc/regex#L4326-L4328